mitodl/mitxpro
# View/edit profile pages

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #201

#### What's this PR do?
Creates "View Profile" and "Edit Profile" pages and forms.
Refactors registration form fields so they are reusable on both the "Edit Profile"  and registration forms

#### How should this be manually tested?
- Register as a new user.  The two registration forms should work as before.
- Go to `/profile`, you should see a page that accurately displays your user info, include profile and legal address fields.
- Go to `profile/edit`, you should see a form that looks like a combination of the two registration forms minus the password field.  It should work correctly (including validation), and you should be redirected back to the profile page on a successful submit.

#### Any background context you want to provide?
I refactored the two registration forms to make the fields reusable on the `Edit Profile` form.

#### Screenshots 
View page (`http://../profile/`)
![Screen Shot 2019-05-24 at 2 22 59 PM](https://user-images.githubusercontent.com/187676/58351019-92a27e80-7e35-11e9-87e2-843cec3c9b38.png)

Edit page (`http://../profile/edit/`)
![Screen Shot 2019-05-24 at 2 20 34 PM](https://user-images.githubusercontent.com/187676/58351040-a1893100-7e35-11e9-85be-29874a01004e.png)

Mobile:
![Screen Shot 2019-05-24 at 2 23 09 PM](https://user-images.githubusercontent.com/187676/58351058-ad74f300-7e35-11e9-97ed-c9d365cb19c0.png)

![Screen Shot 2019-05-24 at 2 20 48 PM](https://user-images.githubusercontent.com/187676/58351053-a948d580-7e35-11e9-81ca-d832c4c8fc72.png)
